spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/discovery/DiscoveryClientRouteDefinitionLocator.java

```diff
@@ -93,8 +94,8 @@ public DiscoveryClientRouteDefinitionLocator(DiscoveryClient discoveryClient,
 			};
 		}
 
-		return Flux.fromIterable(discoveryClient.getServices())
-				.map(discoveryClient::getInstances)
+		return Flux.defer(() -> Flux.fromIterable(discoveryClient.getServices()))
+				.map(discoveryClient::getInstances).subscribeOn(Schedulers.elastic())
```

Comment:
is it Reactor 3.3.0? If so, it's (much) better to use `boundedElastic` here

Reply:
Not this branch. When we merge forward we should update it.